packages/components/src/Table/Header/TableHeader.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import theme from './Header.scss';
+import TableHeaderCell from './TableHeaderCell';
+
+function getHeaderClassName(classNameProvider) {
+	if (classNameProvider && classNameProvider.getForHeader) {
+		return classNameProvider.getForHeader();
+	}
+	return '';
+}
+
+function getHeaderCellClassName(classNameProvider, columnKey) {
+	if (classNameProvider && classNameProvider.getForHeaderCell) {
+		return classNameProvider.getForHeaderCell(columnKey);
+	}
+	return columnKey;
+}
+
+function getHeaderData(rowDataGetter, columnKey) {
+	if (rowDataGetter && rowDataGetter.getHeaderData) {
+		return rowDataGetter.getHeaderData(columnKey);
+	}
+	return columnKey;
+}
+
+function getHeaderComponent(headerRenderer, columnKey) {
+	if (headerRenderer && headerRenderer.getHeaderComponent) {
+		return headerRenderer.getHeaderComponent(columnKey);
+	}
+	return TableHeaderCell;
+}
+
+function getHeaderExtraProps(headerRenderer, columnKey) {
+	if (headerRenderer && headerRenderer.getExtraProps) {
+		return headerRenderer.getExtraProps(columnKey);
+	}
+	return null;
+}
+
+function renderHeaderCell(classNameProvider, rowDataGetter, headerRenderer, columnKey) {
+	const HeaderComponent = getHeaderComponent(headerRenderer, columnKey);
+	const thKey = `th-${columnKey}`;
+	return (
+		<th key={thKey} className={classNames(thKey, theme['tc-table-header-cell'])}>
+			<HeaderComponent
+				key={columnKey}
+				data={getHeaderData(rowDataGetter, columnKey)}
+				className={getHeaderCellClassName(classNameProvider, columnKey)}
+				extra={getHeaderExtraProps(headerRenderer, columnKey)}
+			/>
+		</th>
+	);
+}
+
+/**
+ * This component displays the header of the table.
+ */
+export default function TableHeader({
+	updateHeadNodeRef,
+	columnKeys,
+	classNameProvider,
+	rowDataGetter,
+	headerRenderer,
+}) {
+	const classnames = classNames(
+		'tc-table-header',
+		theme['tc-table-header'],
+		getHeaderClassName(classNameProvider),
+	);
+	return (
+		<thead ref={updateHeadNodeRef} className={classnames}>
+			<tr className={theme['tc-table-header-row']}>
+				{columnKeys.map(col =>
+					renderHeaderCell(classNameProvider, rowDataGetter, headerRenderer, col),
```

Comment:
yes, bad name. I propose to rename it "tableDatagetter".

Reply:
I still don't get it because in the default header component you don"'t use it. What is that ?

Reply:
TableHeader calls the getHeaderData() method in order to get the data corresponding to the column header. This data is then passed to the component which renders the header cell.

Reply:
My question is about `rowDataGetter`

Reply:
yes i understand, rowDataGetter provides data for rows and header. So it is why i propose to rename it to "tableDataGetter". It was called like that because in its initial version it didn't provide data for header.

Reply:
Can you give me an example of what this tableDataGetter would provide ?

Reply:
For the story, i have defined the following tableDataGetter:

	const tableDataGetter = {
	  getElementId(element) {
    	return element.id;
	  },
	  getHeaderData(columnKey) {
		switch (columnKey) {
			case ColumnKey.NAME:
      			return 'Name';
			case ColumnKey.TYPE:
	  			return 'Type';
			case ColumnKey.DESC:
				return 'Description';
			default:
				return columnKey;
		}
	  },
	  getRowData(element, columnKey) {
    	switch (columnKey) {
			case ColumnKey.NAME:
        		return element.name;
			case ColumnKey.TYPE:
				return element.type;
			case ColumnKey.DESC:
				return element.description;
			default:
				return 'No data available!';
 	  	}
	  },
	};